bleak/backends/corebluetooth/CentralManagerDelegate.py

```diff
@@ -146,11 +147,11 @@ async def connect_(self, peripheral: CBPeripheral, timeout=10.0) -> bool:
         self._connection_state = CMDConnectionState.PENDING
         self.central_manager.connectPeripheral_options_(peripheral, None)
 
-        start = time.time()
-        while self._connection_state == CMDConnectionState.PENDING:
-            await asyncio.sleep(0)
-            if time.time() - start > timeout:
-                raise TimeoutError("Failed to connect in %d seconds" % timeout)
+        try:
+            await asyncio.wait_for(self._connection_state_changed.wait(), timeout=timeout)
+        except TimeoutError:
+            logger.debug(f"Connection timed out after {timeout} seconds.")
```

Comment:
We need to call `self.central_manager.cancelPeripheralConnection_(self.connected_peripheral)` here to cancel the pending connection.

Reply:
This explains why, in rare cases, I was seeing a bunch of "Successful" connections all in a row.  Thanks for the heads up.